src/MongoDB.Driver.Legacy/MongoDatabase.cs

```diff
@@ -559,7 +559,7 @@ public virtual MongoCollection<TDefaultDocument> GetCollection<TDefaultDocument>
         public virtual MongoCollection<TDefaultDocument> GetCollection<TDefaultDocument>(
             string collectionName, MongoCollectionSettings collectionSettings)
         {
-            return new MongoCollection<TDefaultDocument>(this, collectionName, collectionSettings);
+            return new MongoCollection<TDefaultDocument>(this, collectionName, collectionSettings, _operationExecutor);
```

Comment:
This is the bug...

The operationExecutor was not being passed down from the MongoDatabase to the MongoCollection.